crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -244,6 +259,17 @@ fn check_statements_require_semicolon(
     })
 }
 
+fn global_const_type_annotation() -> impl NoirParser<UnresolvedType> {
+    ignore_then_commit(just(Token::Colon), parse_type()).map(|r#type| match r#type {
+        UnresolvedType::FieldElement(_) => UnresolvedType::FieldElement(IsConst::Yes(None)),
```

Comment:
I somewhat remember this from the last review, but it still seems odd that this is needed. What were the errors when running without this again?

Reply:
To follow Rust syntax, we can specify `const x: Field = 10`, and `parse_type()` only declares a const if the type annotation is like such `x: const Field`. However, because we use const as an identifier keyword we do not need the redundant `const x: const Field = 10`. Without changing UnresolvedType directly, `parse_type()` will return a `UnresolvedType::FieldElement(IsConst::No(None))`, unless a redundant `const` is specified. I agree, this is a bit odd though, and a kind of a bandaid I wrote while trying to get things working. Perhaps it would be worth it to change consts to only being specified how global consts are currently declared. Or we could add a `parse_type()` for constants (or its internal parsers like `field_type()`) to not check for the const keyword and automatically use `IsConst::Yes`

Reply:
I had actually written a TODO about the redundant const I missed on my final pass, so I'm going to change this. I'm leaning towards removing allowing type annotations like `x: const Field` and just requiring `const x: Field = 10` to follow Rust syntax. What do you think?

Reply:
I think removing them entirely from type annotations may do more harm than good since constness is still currently tracked by the type system. It is different than rust by design currently since we allow for functions to force certain parameters to be const as part of their type - this is how array indexing is currently typed for example. Removing consts from types would either require us to implement another separate system from type checking to check for const-ness and propagate it where required, or require an extension of the current check to manually add const-ness to something.
Either way, I think we avoid the most work and confusion by keeping it in the type system for now. Possibly also removing the 'const' keyword to declare globals since the two concepts should likely be separate anyway. We can (should) also re-explore this when we allow arrays to take non-const indices as well. At that point, const would no longer strictly be required in the type system at all and we can look at either removing it or keeping it for users that do have some reason to force certain optimizations by something being const.